device/edison.go

```diff
@@ -34,86 +35,84 @@ const (
-func initEdison() error {
-	wg := &sync.WaitGroup{}
+type edison struct {
+	*deviceFlasher
+}
 
+func (d *edison) PrepareForFlashing() error {
 	ack := dialogs.YesNoDialog("Would you like to flash your device? ")
 
 	if ack {
-		vm, _, _, _ := vboxDownloadImage(wg, lib.VBoxTemplateEdison, constants.DEVICE_TYPE_EDISON)
-
-		printWarnMessage()
+		return d.deviceFlasher.PrepareForFlashing()
+	}
 
-		for ack {
-			ack = !dialogs.YesNoDialog("Please unplug your edison board. Type 'y/yes' once unpluged. ")
-		}
+	return nil
+}
 
-		for {
-			script := "flashall.sh"
+func (d *edison) Configure() error {
+	for !dialogs.YesNoDialog("Please unplug your edison board. Press yes once unpluged? ") {
+	}
 
-			args := []string{
-				fmt.Sprintf("%s@%s", lib.TemplateUser, lib.TemplateIP),
-				"-p",
-				lib.TemplateSSHPort,
-				constants.TMP_DIR + script,
-			}
+	for {
+		script := "flashall.sh"
 
-			if err := help.ExecStandardStd("ssh", args...); err != nil {
-				fmt.Println("[-] Cannot find mounted Intel edison device, please mount it manually")
+		args := []string{
+			fmt.Sprintf("%s@%s", vbox.VBoxUser, vbox.VBoxIP),
+			"-p",
+			vbox.VBoxSSHPort,
+			constants.TMP_DIR + script,
+		}
 
-				if !dialogs.YesNoDialog("Type 'y/yes' once mounted. ") {
-					fmt.Println("Exiting with exit status 2 ...")
-					os.Exit(2)
-				}
+		if err := help.ExecStandardStd("ssh", args...); err != nil {
+			fmt.Println("[-] Cannot find mounted Intel edison device, please try to re-mount it")
 
-				continue
+			if !dialogs.YesNoDialog("Press yes once mounted? ") {
+				fmt.Println("Exiting with exit status 2 ...")
+				os.Exit(2)
 			}
 
-			break
+			continue
 		}
 
-		fmt.Printf("[+] Stopping virtual machine - Name:%s UUID:%s\n", vm.Name, vm.UUID)
-
-		if err := vbox.Stop(vm.UUID); err != nil {
-			log.Error(err)
-		}
-
-		progress := make(chan bool)
-		go func() {
-			defer close(progress)
-			time.Sleep(120 * time.Second)
-		}()
-
-		help.WaitAndSpin("Configuring edison", progress)
-		<-progress
+		break
+	}
 
+	if err := vbox.Stop(d.vbox.UUID); err != nil {
+		logrus.Error(err)
 	}
 
-	// Config edison
-	config := NewSetDevice(constants.DEVICE_TYPE_EDISON)
+	progress := make(chan bool)
+	go func() {
+		defer close(progress)
+		time.Sleep(120 * time.Second)
+	}()
 
-	help.ExitOnError(config.SetConfig())
+	help.WaitAndSpin("Configuring edison", progress)
+	<-progress
 
-	// Info message
-	printDoneMessageUsb()
+	setConfig()
+
+	fmt.Println(strings.Repeat("*", 100))
+	fmt.Println("*\t\t WARNNING!!  \t\t\t\t\t\t\t\t\t   *")
+	fmt.Println("*\t\t IF YOUR EDISON BOARD IS CONNECTED TO YOUR MACHINE, PLEASE DISCONNECT IT!  \t   *")
+	fmt.Println("*\t\t PLEASE FOLLOW THE INSTRUCTIONS! \t\t\t\t\t\t   *")
+	fmt.Println(strings.Repeat("*", 100))
 
 	return nil
 }
 
-// Uses ifconfig to setup edison interface to be accessable via 192.168.2.2 ip
-func (e *edison) SetConfig() error {
+func setConfig() error {
```

Comment:
__[gometalinter]__
* [gocyclo] cyclomatic complexity 17 of function setConfig() is high (> 15)

<!-- Comment by [SideCI](https://sideci.com) -->